consensus-types/blocks/types.go

```diff
@@ -38,19 +38,22 @@ var (
 
 // BeaconBlockBody is the main beacon block body structure. It can represent any block type.
 type BeaconBlockBody struct {
-	version                int
-	isBlinded              bool
-	randaoReveal           [field_params.BLSSignatureLength]byte
-	eth1Data               *eth.Eth1Data
-	graffiti               [field_params.RootLength]byte
-	proposerSlashings      []*eth.ProposerSlashing
-	attesterSlashings      []*eth.AttesterSlashing
-	attestations           []*eth.Attestation
-	deposits               []*eth.Deposit
-	voluntaryExits         []*eth.SignedVoluntaryExit
-	syncAggregate          *eth.SyncAggregate
-	executionPayload       *engine.ExecutionPayload
-	executionPayloadHeader *engine.ExecutionPayloadHeader
+	version                       int
+	isBlinded                     bool
+	randaoReveal                  [field_params.BLSSignatureLength]byte
+	eth1Data                      *eth.Eth1Data
+	graffiti                      [field_params.RootLength]byte
+	proposerSlashings             []*eth.ProposerSlashing
+	attesterSlashings             []*eth.AttesterSlashing
+	attestations                  []*eth.Attestation
+	deposits                      []*eth.Deposit
+	voluntaryExits                []*eth.SignedVoluntaryExit
+	syncAggregate                 *eth.SyncAggregate
+	executionPayload              *engine.ExecutionPayload
```

Comment:
Can't these be in a single interface `executionPayload` that has different implementations? Otherwise each fork we'll have to add a possibly nil pointer to the `BeaconBlockBody`